### PR TITLE
framework-system-tools: init at unstable-2023-11-14

### PR DESCRIPTION
**Pull request by kloenk**
## Description of changes
init https://github.com/FrameworkComputer/framework-system/tree/main

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

CC: @leona-ya

### CODE REVIEW

**Comment by yu-re-ka** [comment]
Please check CONTRIBUTING.md: 

> ### Syntax
> 
> - Use 2 spaces of indentation per indentation level in Nix expressions, 4 spaces in shell scripts.
> - Do not use tab characters[....]

File: pkgs/by-name/fr/framework-system-tools/package.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    gnumake
+    pkg-config
+  ];
+
+  buildInputs = [
+    udev
+  ];
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```
Comment (yu-re-ka) [inline pkgs/by-name/fr/framework-system-tools/package.nix:37]: Would always be better if we can convince upstream to switch to crates.io releases. Maybe ask them nicely, but it should not block the merging of this PR.

File: pkgs/by-name/fr/framework-system-tools/package.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    outputHashes = {
+      "smbios-lib-0.9.1" = "sha256-3L8JaA75j9Aaqg1z9lVs61m6CvXDeQprEFRq+UDCHQo=";
+      "uefi-0.20.0" = "sha256-/3WNHuc27N89M7s+WT64SHyFOp7YRyzz6B+neh1vejY=";
+    };
+  };
+
+  meta = with lib; {
+    description = "Rust libraries and tools to interact with the  framework system.";
```
Comment (yu-re-ka) [inline pkgs/by-name/fr/framework-system-tools/package.nix:45]:
```suggestion
    description = "Rust libraries and tools to interact with the framework system.";
```

File: pkgs/by-name/fr/framework-system-tools/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    owner = "FrameworkComputer";
+    repo = "framework-system";
+    inherit rev;
+    hash = "sha256-qDtW4DvY19enCfkOBRaako9ngAkmSreoNWlL4QE2FAk=";
+  };
+
+  nativeBuildInputs = [
+    cargo
+    rustc
+    gnumake
```
Comment (yu-re-ka) [inline pkgs/by-name/fr/framework-system-tools/package.nix:28]: is this not in stdenv?

File: pkgs/by-name/fr/framework-system-tools/package.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  src = fetchFromGitHub {
+    owner = "FrameworkComputer";
+    repo = "framework-system";
+    inherit rev;
+    hash = "sha256-qDtW4DvY19enCfkOBRaako9ngAkmSreoNWlL4QE2FAk=";
+  };
+
+  nativeBuildInputs = [
+    cargo
```
Comment (yu-re-ka) [inline pkgs/by-name/fr/framework-system-tools/package.nix:26]: cargo and rustc are passed automatically by buildRustPackage

Comment (Kloenk) [reply pkgs/by-name/fr/framework-system-tools/package.nix:26]: Ah yes thanks. this is a forgotten leftover from trying to also build the efi file, which failed

File: pkgs/by-name/fr/framework-system-tools/package.nix (hunk 1 above)
Comment (Kloenk) [reply pkgs/by-name/fr/framework-system-tools/package.nix:30]: Upstream seems to be working on upstreaming their changes, but looks like still ongoing
https://github.com/FrameworkComputer/framework-system/blob/main/framework_lib/Cargo.toml#L61

**Review by sivizius — APPROVED** [review]
LGTM 🚀

**Comment by NickCao** [comment]
Looks like this is a duplicate of https://github.com/NixOS/nixpkgs/pull/267578, we should merge them.

**Pull request merged**